LibreHardwareMonitorLib/Hardware/RyzenSMU.cs

```diff
@@ -76,20 +76,30 @@ internal class RyzenSMU
                 // Zen 3.
                 0x00380805, new Dictionary<uint, SmuSensorType>
                 {
-                    // TDC and EDC don't match the HWiNFO values
-                    //{ 15, new SmuSensorType { Name = "TDC", Type = SensorType.Current, Scale = 1 } },
-                    //{ 21, new SmuSensorType { Name = "EDC", Type = SensorType.Current, Scale = 1 } },
+                    { 3, new SmuSensorType { Name = "TDC", Type = SensorType.Current, Scale = 1 } },
+                    // TODO: requires some post-processing
+                    // see: https://gitlab.com/leogx9r/ryzen_smu/-/blob/master/userspace/monitor_cpu.c#L577
+                    // { 9, new SmuSensorType { Name = "EDC", Type = SensorType.Current, Scale = 1 } },
```

Comment:
Anything you want to do for this still? I guess some kind of ScaleFunc could be added.

Reply:
It's not that easy, unfortunately - the final value requires knowing the values of a bunch of other metrics, and that's going to involve a pretty big refactoring.

Reply:
What is needed exactly? The sensor should already have the hardware (CPU) for example.

Reply:
A bunch of other SMU readings, specifically core C0 residency info and such.

Reply:
(feel free to check the link in the source comment, I'm on my phone right now so details might be a bit off)

Reply:
I see, if needed that can be added later :) Rest seems good, apart from removing (Effective) to follow the others.

Reply:
The "Effective" bit is important, it's not the same clock reading as instant clock samples.

Reply:
True, the others must probably also be renamed to Effective then. You want to take that? 🙂

Reply:
What others?

Reply:
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/c5609ef7caa77e3f0deb101678569a29a1cfe38d/LibreHardwareMonitorLib/Hardware/RyzenSMU.cs#L42-L45

Reply:
No, those are temperature readings, not clocks.